libavcodec/hwaccel.h

```diff
@@ -24,6 +24,7 @@
 
 
 #define HWACCEL_CAP_ASYNC_SAFE      (1 << 0)
+#define HWACCEL_CAP_INTERNAL_ALLOC  (1 << 1)
```

Comment:
better change to a better name like HWACCEL_CAP_ FORMAT_CHANGE_ON_THE_FLY and add some comment.